segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/ServerConnectionInboundHandler.java

```diff
@@ -53,7 +53,15 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
         if (requestProcessor == null) {
             throw new IllegalStateException("No command processor set for connection");
         }
-        cmd.process(requestProcessor);
+        try {
+            cmd.process(requestProcessor);
+        } catch (Throwable ex) {
+            // Release buffers in case of an unhandled exception.
+            if (cmd instanceof WireCommands.ReleasableCommand) {
```

Comment:
What about adding an Utility method
Wire command.ReleasableCommand#safeRelease(cmd) ?

This way we can save from writing **instanceof** that most of the times looks like a code smell

Reply:
I thought about that, but we are actually aiming to eliminate Netty from our Wire Protocol altogether due to poor performance in some cases, so we might not need to release anything afterall. That's also a reason why we're consolidating our buffers under `BufferView` and not exposing `ByteBuf` in our internal/external APIs.